tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs

```diff
@@ -0,0 +1,244 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Utils;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using NATS.Client.Core;
+using NATS.Client.JetStream;
+using NATS.Client.JetStream.Models;
+namespace Aspire.Hosting.Nats.Tests;
+
+public class NatsFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyNatsResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var nats = builder.AddNats("nats");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{nats.Resource.Name}"] = await nats.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddNatsClient(nats.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var natsConnection = host.Services.GetRequiredService<INatsConnection>();
+
+            var rtt = await natsConnection.PingAsync(token);
```

Comment:
Should we do something a little more than ping it? Can we send and consume a message?

Reply:
I will do.

Reply:
I don't think this was resolved.